scRNA-seq-advanced/03-dataset_integration.Rmd

```diff
@@ -502,66 +702,104 @@ harmony_pcs <- harmony::HarmonyMatrix(
+  guides(colour = guide_legend(override.aes = list(size = 3, alpha = 1))) 
+```
+Similar to the `fastMNN` results, the non-tumor cell types do appear to have been well-integrated across batches, but that one library does stand out!
+We can see that 
+
+Now is a good time to remember we actually have two cell type variables: `celltype_broad` and `celltype_fine`, where the latter contains finer-grained Tumor characterizations based on differentiation stage.
+Let's see what we observe if we color the UMAP instead by the finer-grained cell types:
+
+```{r harmony umap celltypes fine}
+scater::plotReducedDim(merged_sce,
+                       dimred = "harmony_UMAP",
+                       # color by FINE celltypes
+                       colour_by = "celltype_fine",
+                       point_size = 0.5,
+                       point_alpha = 0.1, 
                        other_fields = "sample") +
-  ggplot2::ggtitle("UMAP after integration with harmony") +
-  # Modify the ggplot2 guides so that the legend points are larger and easier to see:
-  ggplot2::guides(colour = ggplot2::guide_legend(override.aes = list(size = 3, alpha = 1))) +
-  ggplot2::facet_wrap(~sample, nrow=2)
+  ggtitle("UMAP after integration with harmony") +
+  guides(colour = guide_legend(override.aes = list(size = 3, alpha = 1))) +
+  facet_wrap(~sample)
+  
+```
+
+It looks as though `SCPCL000481` may indeed be comprised of relatively distinct cell types, which we can check with a barplot:
+
+<!-- LATE BREAKING ADDITION IN NEED OF REVIEW/DISCUSSION, including where to put this plot? Should I make this plot? -->
+```{r count celltypes fine}
+# Create a table of counts sample and celltype_fine combinations
+sample_celltype_table <- colData(merged_sce) %>%
+  as.data.frame() %>%
+  dplyr::count(sample, celltype_fine) 
+
+
+sample_celltype_table
+
+# Let's look at just tumor cells:
+sample_celltype_table %>%
+  dplyr::filter(stringr::str_detect(celltype_fine, "Tumor")) %>%
+  ggplot(aes(x = sample, y = n, fill = celltype_fine)) +
+  geom_col(position = "dodge")
 ```
 
-How do you think these methods' (`fastMNN` and `harmony`) performances compare on these data?
+Indeed, we see that `SCPCL000481` has a substantial `Tumor_Myocyte-A` population that the other libraries do not have, and the two libraries from the same patient show similar trends. 
+
+
+What similarities and differences between `fastMNN` and `harmony` integration do you notice?
+How do you think these results compare?
+
+### Export 
 
-Let's now save our final SCE object to a file.
-We'll also specify the argument `compress = "bz2"`, which saves the file with an added layer of compression. 
-This file is otherwise quite large!
+Finally, we'll export the final SCE object with both `fastMNN` and `harmony` integration to a file.
 
 ```{r save integration}
-readr::write_rds(merged_sce, integrated_sce_file, compress = "bz2")
+readr::write_rds(merged_sce, 
+                 integrated_sce_file)
```

Comment:
I would leave compression, but use "gz" which should be faster at the expense of only a little disk space. 

It still takes ~2 minutes (by system.time, plus the time to actually write the file, which happened later and still blocked interaction), but 330MB vs 1.5GB seems a worthwhile savings. 

(bz2 got it half again as small, but took ~3 minutes... but bz2 also takes longer to open, so probably not worth the effort)
We can talk about it while the saving happens!

(Noting that didn't compress in the setup scripts... gonna file a PR to add that.)

```suggestion
readr::write_rds(merged_sce, 
                 integrated_sce_file,
                 compress = "gz")
```